laravel/the-basics/databases.html.md

```diff
@@ -0,0 +1,198 @@
+<p></p>
+
+## _[Fly .internal Address](/docs/reference/private-networking/#fly-internal-addresses)_
+
+In this guide, you&#39;ll be connecting your Laravel application with various database applications running in Fly.io. A very important concept in connecting Fly.io applications with each other is [Private Networking](/docs/reference/private-networking).
+
+<aside class="callout">
+How do you connect your Laravel application running in Fly.io with a database application running in Fly.io?
+</aside> 
+
+<b>The Answer: Use the database application&#39;s [Fly .internal Address](/docs/reference/private-networking/#fly-internal-addresses):</b>
+<ol>
+<li>Get the database application&#39;s app-name from its `fly.toml`</li>
+<li>Generate the pattern for your database application&#39;s  Fly .internal Address: `"<app-name>.internal"`</li>
+<li>Use the Fly .internal Address retrieved above in your Laravel application&#39;s `fly.toml` configuration for connecting with the database</li>
+</ol>
+
+
+## _MySQL in Fly.io_
+You can start with a relational-database classic: [MySQL](https://www.mysql.com/); you can even run it as a Fly App.
+
+### Setup a MySQL application in Fly.io
+Follow this [MySQL guide](/docs/app-guides/mysql-on-fly/) to get up and running with your MySQL application in Fly.io.
+
+### Get MySQL application's Fly .internal Address
+Open your MySQL application's `fly.toml` as a reference, and from there get its [Fly .internal address](/docs/laravel/the-basics/databases/#fly-internal-address). 
+As an example, your MySQL Fly .internal Address would be `"mysql-app-name.internal"`, given a `fly.toml` configuration below:
+
+```toml
+app = "mysql-app-name"
+
+[env]
+  MYSQL_DATABASE = "<database-name>"
+  MYSQL_USER =  "<database-user>"
+```
+
+### Connect from a Laravel application in Fly.io
+First you'll have to revise the `[env]` configuration in your Laravel application&#39;s `fly.toml` file to connect with your MySQL application's [Fly .internal address](/docs/laravel/the-basics/databases/#get-mysql-application-s-fly-internal-address) and MYSQL_DATABASE:
+
+```toml
+[env]
+  APP_ENV = "production"
+  DB_CONNECTION = "mysql"
+  DB_HOST = "<MYSQL Fly .internal address>"
+  DB_DATABASE= "<MYSQL_DATABASE>"
+```
+
+Then, set up the database username and password through [Fly Secrets](https://fly.io/docs/flyctl/secrets/):
+
+```cmd
+fly secrets set DB_USERNAME=<MYSQL_USER> DB_PASSWORD=<MYSQL_PASSWORD>
+```
+
+Finally deploy your changes with
+
+```cmd
+fly deploy 
+```
+
+
+
+### Connect  from a local environment
+
+The MySQL instance you spun up in Fly.io &quot;[is closed to the public internet](https://fly.io/docs/reference/machines/#notes-on-networking)&quot;, and can only be accessed by another application found in your Fly.io organization&#39;s private network. You&#39;ll need a way to tunnel into the network, and finally connect to your MySQL instance.
```

Comment:
These URL's can usually have the `https://fly.io` portion removed so they're relative URL's, like `/docs/reference/machines/#notes-on-networking` - This helps to keep you in the local/preview environments so you don't accidentally jump into the production environment when clicking around.